include/zephyr/bluetooth/conn.h

```diff
@@ -1063,6 +1063,16 @@ struct bt_conn_cb {
  */
 void bt_conn_cb_register(struct bt_conn_cb *cb);
 
+/** @brief Unregister connection callbacks.
+ *
+ *  Unregister the state of connections callbacks.
+ *
+ *  @param cb Callback struct point to memory that remains valid.
+ *
+ *  @return Zero on success or negative error code otherwise.
+ */
+int bt_conn_cb_unregister(struct bt_conn_cb *cb);
```

Comment:
I think we should expand at least one of our tests (possibly babblesim (bsim) tests, or a unit test) to run and test this functionality.

Reply:
They added a test, could you re-review?